br/pkg/storage/ks3.go

```diff
@@ -673,9 +673,9 @@ func (rs *KS3Storage) Create(ctx context.Context, name string, option *WriterOpt
 		s3Writer.wg.Add(1)
 		go func() {
 			_, err := up.Upload(upParams)
-			err1 := rd.Close()
 			if err != nil {
-				log.Warn("upload to s3 failed", zap.String("filename", name), zap.Error(err), zap.Error(err1))
+				log.Warn("upload to ks3 failed", zap.String("filename", name), zap.Error(err))
+				_ = rd.CloseWithError(err)
```

Comment:
so there's no need to close rd when Upload returns no err?

Reply:
Yes, I see PipeReader does not release resources in Close.

Reply:
better call it or explain in the comments, looks like a bug

Reply:
https://github.com/pingcap/tidb/pull/48163/commits/3a0fd86ae8b9d9e0021f2c135bfa45c4568fcbaf